kv.go

```diff
@@ -158,26 +158,27 @@ func decodeHook(fromType reflect.Type, toType reflect.Type, data interface{}) (i
 
 			return dataOutput, nil
 		} else if fromType.Kind() == reflect.String {
-			buffer := bytes.NewBuffer([]byte(data.(string)))
-			reader, err := gzip.NewReader(buffer)
-			if err != nil {
-				decodedData, err := base64.StdEncoding.DecodeString(data.(string))
-				if err != nil {
-					return nil, err
-				}
-				return decodedData, nil
-			}
-			defer reader.Close()
-			decodedData, err := ioutil.ReadAll(reader)
-			if err != nil {
-				return nil, err
-			}
-			return decodedData, nil
+			return gzipReader(data)
 		}
 	}
 	return data, nil
 }
 
+func gzipReader(data interface{}) (interface{}, error) {
```

Comment:
`interface{}` in arg and return seems a bit *risky* if you wanna reuse that func. I would do the cast before and return `io.Reader, error` 👼